src/sudo/pipeline/list.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::Path;
+
+use crate::{
+    cli::{SudoAction, SudoOptions},
+    common::Error,
+    log::auth_warn,
+    pam::CLIConverser,
+    sudo::{pam::PamAuthenticator, SudoersPolicy},
+    sudoers::{Authorization, ListRequest, Policy, Request},
+    system::{timestamp::RecordScope, Process, User},
+};
+
+use super::{AuthPlugin, Pipeline, PolicyPlugin};
+
+impl Pipeline<SudoersPolicy, PamAuthenticator<CLIConverser>> {
+    pub(in crate::sudo) fn run_list(mut self, cmd_opts: SudoOptions) -> Result<(), Error> {
```

Comment:
`Pipeline::run_list` is different enough from the other two `run` methods that it warrants being in its own module, IMO. that way, any helper function / method that it uses, and the other two `run`s won't, can be kept private to that module (see the updated `list.rs` file after a refactor for readability). conversely, code shared between `run_list` and the other `run` methods can go in the parent module (`pipeline.rs`) so that it can be seen by all its users (e.g. see the new `Pipeline::auth_and_update_record_file` method)

^ that approach also avoids files getting too large (`pipeline/list.rs` is ~200 lines; `pipeline.rs` is ~300 lines). I personally prefer to have several self contained files, e.g. one file per feature with an inline `mod tests`, but I know some people find hard to navigate code bases that have many files. (TBH, navigation shouldn't be an issue if you are familiar with LSP's "go to definition / references / type / implementations / etc." actions)